blueprints/controllers/ikea_e1744/ikea_e1744.yaml

```
@@ -215,11 +241,9 @@ action:
           # repeat until the automation is restarted ie. when the corresponding stop message is received
           - repeat:
               sequence: !input action_rotate_left
-              until: '{{ not rotate_left_loop }}'
+              while: '{{ rotate_left_loop }}'
```

```suggestion
              until: '{{ not rotate_left_loop }}'
```

We have to keep using a post-test conditional loop to make sure the custom `rotate` action gets executed at least once, if the `rotate_left_loop` is set to `false`. In this case, the `rotate` action needs to be performed only once.

---

I don't think there is a difference in how it works. both check conditions every loop. But I don't like a negative (not) if it's not needed :)

---

Actually there is:
> ### REPEAT UNTIL
> This form accepts a list of conditions that are evaluated **after** each time the sequence is run. Therefore **the sequence will always run at least once**. The sequence will be run until the condition(s) evaluate to true.



https://www.home-assistant.io/docs/scripts/#repeat-until

---

I see, sorry about that.

But in a while loop, you only want to execute if the event is still a rotate event right? So if we evaluate before the start of the loop and the event is no longer a rotate-event. it should stop there, no?